src/Kurrent.Client/Core/Serialization/MessageSerializer.cs

```diff
@@ -0,0 +1,151 @@
+using System.Diagnostics.CodeAnalysis;
+using EventStore.Client;
+using EventStore.Client.Diagnostics;
+using EventStore.Client.Serialization;
+using Kurrent.Client.Core.Serialization;
+
+namespace EventStore.Client.Serialization;
+
+using static ContentTypeExtensions;
+
+public interface IMessageSerializer {
+	public EventData Serialize(Message value, MessageSerializationContext context);
+
+#if NET48
```

Comment:
`NotNullWhen` and other similar attributes compile fine for .NET 4.8 target using the latest .NET SDK.